homeassistant/helpers/httpx_client.py

```diff
@@ -0,0 +1,194 @@
+"""Helper for httpx."""
+from functools import partial
+import sys
+from typing import Any, Callable, cast
+
+import httpx
+
+from homeassistant.const import EVENT_HOMEASSISTANT_CLOSE, __version__
+from homeassistant.core import Event, callback
+from homeassistant.helpers.frame import warn_use
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.loader import bind_hass
+from homeassistant.util.async_ import run_callback_threadsafe
+
+DATA_CLIENT = "httpx_client"
+DATA_CLIENT_NOVERIFY = "httpx_client_noverify"
+DATA_ASYNC_CLIENT = "httpx_async_client"
+DATA_ASYNC_CLIENT_NOVERIFY = "httpx_async_client_noverify"
+SERVER_SOFTWARE = "HomeAssistant/{0} httpx/{1} Python/{2[0]}.{2[1]}".format(
+    __version__, httpx.__version__, sys.version_info
+)
+USER_AGENT = "User-Agent"
+
+
+@callback
+@bind_hass
+def async_get_async_client(
+    hass: HomeAssistantType, verify_ssl: bool = True
+) -> httpx.AsyncClient:
+    """Return default httpx AsyncClient.
+
+    This method must be run in the event loop.
+    """
+    key = DATA_ASYNC_CLIENT if verify_ssl else DATA_ASYNC_CLIENT_NOVERIFY
+
+    hass.data.setdefault(key, async_create_async_httpx_client(hass, verify_ssl))
+
+    return cast(httpx.AsyncClient, hass.data[key])
```

Comment:
This is creating one every time even if it is unused. Better break it up:

```suggestion
    client: Optional[httpx.AsyncClient] = hass.data.get(key)
    
    if client is None:
        client = hass.data[key] = async_create_async_httpx_client(hass, verify_ssl)

    return client
```